proposals/target-typed-new.md

```diff
@@ -0,0 +1,100 @@
+
+# Target-typed `new` expressions
+
+* [x] Proposed
+* [x] [Prototype](https://github.com/alrz/roslyn/tree/features/implicit-new)
+* [ ] Implementation
+* [ ] Specification
+
+## Summary
+[summary]: #summary
+
+Do not require type specification for constructors when the type is known. 
+
+## Motivation
+[motivation]: #motivation
+
+Allow field initialization without duplicating the type.
+```cs
+Dictionary<string, List<int>> field = new() {
+    { "item1", new() { 1, 2, 3 } }
+};
+```
+Allow omitting the type when it can be inferred from usage.
+```cs
+XmlReader.Create(reader, new() { IgnoreWhitespace = true });
+```
+
+## Detailed design
+[design]: #detailed-design
+
+The *object_creation_expression* syntax would be modified to make the *type* optional when parentheses are present. This is required to address the ambiguity with *anonymous_object_creation_expression*.
+```antlr
+object_creation_expression
+    : 'new' type? '(' argument_list? ')' object_or_collection_initializer?
```

Comment:
It is not clear how or when the `object_or_collection_initializer` is bound. I presume it is bound as part of the conversion (i.e. late), but that errors within it do not affect whether or not the given constructor is inferred. That should be made explicit.

Reply:
> that errors within it do not affect whether or not the given constructor is inferred

we don't bind the initializer when we try to create the conversion, it only happens after the conversion succeeded. This is mentioned in the next line:

>Target of the initialization is determined after the conversion succeeded.